parlai/agents/bert_classifier/bert_classifier.py

```diff
@@ -106,26 +110,31 @@ def add_cmdline_args(parser):
 
     @staticmethod
     def dictionary_class():
+        """Determine the dictionary class."""
         return BertDictionaryAgent
 
-    def _upgrade_opt(self, opt):
-        model_opt = opt['model_file'] + '.opt'
-        if not os.path.isfile(model_opt):
-            return
-        old_opt = load_opt_file(model_opt)
-        if 'add_cls_token' not in old_opt:
-            # old model, make this default to False
+    @classmethod
+    def upgrade_opt(cls, opt_on_disk):
+        """Upgrade opts from older model files."""
+        super(BertClassifierAgent, cls).upgrade_opt(opt_on_disk)
+
+        # 2019-06-25: previous versions of the model did not add a CLS token
+        # to the beginning of text_vec.
+        if 'add_cls_token' not in opt_on_disk:
             warn_once('Old model: overriding `add_cls_token` to False.')
-            opt['add_cls_token'] = False
-        return
+            opt_on_disk['add_cls_token'] = False
```

Comment:
nice

Reply:
is opt_from_disk better? opt_on_disk sounds subtly like you're updating disk.

Reply:
i like opt_from_disk